src/components/shared/ProgressionControl.tsx

```diff
@@ -11,7 +11,7 @@ type ProgressionControlChildProps =
 type ProgressionControlChild = React.ReactElement<ProgressionControlChildProps>;
 interface ProgressionControlProps {
     children: ProgressionControlChild;
-    onPage: number | number[];
+    onPage: { [key in Module]?: number | number[] };
```

Comment:
I wonder if you could simplify the type by just always using `number[]` even if it's just a single one.